File: Assignment/3학년 1학기/Java/기말고사/클래스 설계/Oak.java
package Project1;

public class Oak extends Character {

	public Oak() {
		this.name = "None";
		this.brood = "오크";
	}
	
	public Oak(String name)
	{
		this.name = name;
		this.brood = "오크";
	}
	
	public void ActiveSkill()
	{
		System.out.println("돌진");
	}
	
	public void PassiveSkill()
	{
		System.out.println("근력 상승");
	}

}

File: Assignment/2학년 1학기/C++/중간고사/Oak.cpp
#include "Oak.h"

// 생성자에서 초기화한 값은 부모 클래스에게 넘겨줌
Oak::Oak(const char* _name) : Character(_name) {
	memset(brood, 0, sizeof(brood));
	memset(job, 0, sizeof(job));
}

Oak::~Oak() {

}

// 쓰기 접근자
void Oak::SetBrood(const char* _brood) {
	strcpy(brood, _brood);
}

void Oak::SetJob(const char* _job) {
	strcpy(job, _job);
}

// 읽기 접근자
const char* Oak::GetBrood() {
	return brood;
}

const char* Oak::GetJob() {
	return job;
}